### DISCUSSION TITLE
Running langium:generate and build lead to an error that the generated/ast.js has no exported member 'Model'

### DISCUSSION THREAD

**Discussion by bogdanbiv** [eclipse-langium/langium#1630] [Q&A] [answered]
Running langium:generate with no errors:
`$ npm run langium:generate
> marbles@0.0.1 langium:generate
> langium generate
Reading config from /home/bogdanbiv/git/langium-marbles/marbles/langium-config.json
Writing generated files to /home/bogdanbiv/git/langium-marbles/marbles/src/language/generated
Writing textmate grammar to /home/bogdanbiv/git/langium-marbles/marbles/syntaxes/marbles-diagrams.tmLanguage.json
Writing monarch grammar to /home/bogdanbiv/git/langium-marbles/marbles/src/syntaxes/marbles-diagrams.monarch.ts
[16:16:41] Langium generator finished successfully in 249ms`

Yet, build does not work:
`$ npm run build
> marbles@0.0.1 build
> tsc -b tsconfig.src.json && node esbuild.mjs
src/cli/generator.ts:1:15 - error TS2305: Module '"../language/generated/ast.js"' has no exported member 'Model'.
1 import type { Model } from '../language/generated/ast.js';
                ~~~~~
src/cli/generator.ts:14:74 - error TS18046: 'greeting' is of type 'unknown'.
14         ${joinToNode(model.greetings, greeting => `console.log('Hello, ${greeting.person.ref?.name}!');`, { appendNewLineIfNotEmpty: true })}
                                                                            ~~~~~~~~
src/cli/main.ts:1:15 - error TS2305: Module '"../language/generated/ast.js"' has no exported member 'Model'.
1 import type { Model } from '../language/generated/ast.js';`

How could I start from a clean slate?

**Comment by msujew** [comment, marked as answer]
Hey @bogdanbiv,

the generated code is supposed to show how to setup a CLI for the generated default language. Since it uses types from the default language (i.e. the "Hello World" language), it will fail to compile once you start changing the grammar. If you have no need of a CLI, you can just remove the code. Note that the yeoman generator asks you whether you want to setup a CLI for your language.

**Reply by bogdanbiv** [reply]
it's unclear to me if I would need CLI support in the future. For now, I'm just experimenting, so it's definitely not needed. Thank you for your helping hand
